### DISCUSSION TITLE
Diskussion über die Funktionen im Funktionen-R-Skript 1.R

### DISCUSSION THREAD

**Discussion by DeepRamful** [DeepRamful/Gruppenarbeit-GitHub#33] [General]
Welche Funktionen sind eurer Meinung nach sinnvoll und
notwendig ?

**Comment by vafkoo** [comment]
Funktion D die zwischen einer metrischen und einer
dichotomen Variablen berechnet und ausgibt, normaleweise es ist unmöglich die korrelation mit den (Ja/Nein) Parameter zu berechnen aber diese funktion gibt dieser Parameter den Werten 1 und 0. Mit dieser Funktion kann man die Korellation mit dichtonomen Variablen einfach berrechnen ohne Extra zeit zu verwenden, finde es geil.

**Reply by DeepRamful** [reply]
Du hast gut getan, das Ja/Nein in 1/0 umzuwandeln.

**Comment by DeepRamful** [comment]
Die Funktion (f), die ein Mosaikdiagramm für 3 kategoriale Variablen erstellt, ist nicht so einfach zu interpretieren, aber machbar. Aber wenn wir eine weitere kategoriale Variable hinzufügen, wäre es meiner Meinung nach sehr schwer zu interpretieren.
Deshalb ist es besser, mit nur 3 Variablen zu arbeiten.

**Reply by QuangHuy14899** [reply]
ja, du hast Recht, 3 Variablen sind perfekt, eine mehr ist schwieriger zu interpretieren

**Comment by QuangHuy14899** [comment]
Die Funktion a enthält verschiedene Funktionen, die zur Berechnung verwendet werden. Mit der Funktion a bekommt man alle wichtige Lagemaße einer metrischen Variable, wie Mittelwert, Median,  minimale und maxima Wert, Varianz,... es sparte Zeit bei der Berechnung. 
Funktion b is besonders nützlich, wenn wir Barplot darstellen, weil die Ergebnisse von Funktion b relative und absolute Häufigkeit haben. Achtung, die Funktion b kann nur für kategoriale Variablen verwendet werden.

**Reply by DeepRamful** [reply]
Ja, gut. Du hättest auch die Funktion summary() verwenden können, um einige Statistiken für (a) zu erhalten.
Ich denke auch, dass wir für die Visualisierung dieser Statistiken keine Funktion und ihre Outputs brauchen, denn wenn man ein Histogramm oder einen Boxplot erstellt, werden alle diese Statistiken bereits intern berechnet.

**Comment by Jungangyoung** [comment]
Funktion c finde ich sinnvoll, weil die Kodierung notwendig in diesem Fall ist. 
Die Funkion c hat eine Kodierung für Studienfach. ich denke, dass eine kodierung hier sein muss, weil Studienfach eine nominalskala ist, und wenn man eine Korrelation (Zahl) zwischen nominalskala(keine Zahl) und ordinalskala(Zahl) berechnen will, muss alles als Zahlen ersetzt werden, deswegen mach ich Kodierung, ein Nom zu eine einfache Zahl. 
  Aber die Kodierung hat auch Nachteile, zum Beispiel die Werte der Korrelationen können sehr nah sein, davon kann man nicht einfach sehen, die Unterschied der verschiedenen Relationen, in diesem Fall ist die Korrelation nicht stark genug, eine Zusammenhang zu zeigen.
   Ungenauigkeit ist in Statistik unvermeidlich, so hab ich davon gelernt.

**Reply by QuangHuy14899** [reply]
ja klar, ich finde, dass die Kodierung trotz der Ungenauigkeit sehr hilfreich für die grafische Erstellung ist.
